source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,443 @@
+to read data). GraphQL modules must declare mutations by extending the top-level
+GraphQL `Mutation` type. You can read more about Mutations in the
+[GraphQL’s Mutations documentation](https://graphql.org/learn/queries/#mutations).
+
+```js
+// …
+module.exports = {
+  // …
+  Mutation: {
+    incrementProductCounter(_, { sku, incrementValue = 1 }) {
+      counters.set(sku, currentValueOf(sku) + incrementValue);
+      return {
+        success: true
+      };
+    }
+  }
+};
+```
+
+In this resolver, the first parameter is unused (because in a top level resolver
+no earlier resolvers have been run so there are no _parent data_). The second
+parameter contains arguments passed to the mutation. These parameters are the
+`(sku: String!, incrementValue: Int)` part of the schema definition declared
+earlier:
+
+```graphql
+extend type Mutation {
+  incrementProductCounter(sku: String!, incrementValue: Int): MutationSuccess
+}
+```
+
+The resolver uses the mandatory `sku` argument to fetch the current counter
+value from its local state and defines a new value incremented of the
+`incrementValue` optional argument. If the `incrementValue` has not been
+provided in the GraphQL request, it defaults to `1`.
+
+Finally, the resolver returns a value matching the `MutationSuccess`
+Front-Commerce type which in this example is always successful.
+
+<blockquote class="info">
+  ProTip™: you can debug the data passed in a resolver using `console.log` to
+  view its shape in your server console shell output. To do so, a convenient
+  thing to know is that `console.log` returns a falsy value. Hence, in the
+  resolver above you could debug the `sku` value by updating the resolver as
+  below:
+
+```diff
+-clicksCounter: ({sku}) => currentValueOf(sku),
++clicksCounter: ({sku}) => console.log(sku) || currentValueOf(sku),
+```
+
+</blockquote>
+
+**Awesome!** Your GraphQL server now has a new feature, let’s use it!
+
+### Query your new graph
+
+Try to execute the previous query again in your GraphQL playground:
+
+```graphql
+# http://localhost:4000/playground
+{
+  product(sku: "WH09") {
+    sku
+    clicksCounter
+  }
+}
+```
+
+You must now have a real value returned!
+
+```json
+{
+  "data": {
+    "product": {
+      "sku": "WH09",
+      "clicksCounter": 0
+    }
+  }
+}
+```
+
+Let’s increment it by sending a Mutation to our server:
+
+```graphql
+# http://localhost:4000/playground
+mutation {
+  incrementProductCounter(sku: "WH09", incrementValue: 2) {
+    success
+  }
+}
+```
+
+If it was successful, the above query must now return `2` as the `clicksCounter`
+value.
+
+### Asynchronous resolvers
+
+In the previous example resolvers were synchronous, meaning that they returned a
+value that was immediately available. Most of the time, your resolvers will
+fetch data from a remote service and return them asynchronously using a
+`Promise`.
+
+Returning `Promise` from resolvers is supported out of the box. We encourage you
```

Comment:
I don't see the value of the `We encourage you to experiment with it.` sentence.

There's quite no use of experimenting with hand made Promise here. I'd be better to talk about experimenting with an API imo.